### PR TITLE
Implements SSL-EY

### PR DESCRIPTION
**Pull request by jameschapman19**
Adding SSL-EY from (https://arxiv.org/abs/2310.01012, https://neurips.cc/virtual/2023/80864)

For the moment I've implemented the simple version but it should in principle plug into multicrop like SWAV.

Will have a stab at running the shorter benchmarks shortly.

### CODE REVIEW

**Comment by vturrisi** [comment]
hey @jameschapman19 thanks for contributing. I'll try to start checking your code soon.

File: scripts/pretrain/imagenet-100/ssley.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
+defaults:
+  - _self_
+  - augmentations: ssley.yaml
```
Comment (vturrisi) [inline scripts/pretrain/imagenet-100/ssley.yaml:3]: You already have the augmentations below, either remove this or move the augmentations to a separate file (both are fine)

**Comment by jameschapman19** [comment]
@vturrisi - just working through ironing things out perhaps a premature PR I just wanted to get a hang of the CI. Will give you a shout when I'm passing more of these.

File: scripts/pretrain/imagenet-100/ssley.yaml (hunk 1 above)
Comment (jameschapman19) [reply scripts/pretrain/imagenet-100/ssley.yaml:3]: Good spot - a heads up I think this is also the case for scripts/pretrain/imagenet-100/vicreg.yaml in that case.

This method should plug in wherever VICReg is used. In many ways the benefit of the loss function is it's got a lot of the nice properties of VICReg but without needing to tune the three loss function parameters.

**Comment by vturrisi** [comment]
@jameschapman19 no worries, take your time :) and ping me when you are done

File: scripts/pretrain/imagenet-100/ssley.yaml (hunk 1 above)
Comment (vturrisi) [reply scripts/pretrain/imagenet-100/ssley.yaml]: No idea why VICReg is like that haha. I'm fine with having a separate file for the augmentations or having it merged with the main file. Should also prob fix VICReg after that.
